[Fix bug 1247219] UI fixes for new group settings

comzeradd opened the pull request:
* Minimize space on create group modal
* Use even space on no pending invitations text
* Minimize space on group types descriptions
* Add markdown informative text on terms
* Fix group access criteria toggle

comzeradd commented:
@johngian r?

johngian commented on mozillians/jinja2/groups/includes/group_edit_invitation.html:
```diff
@@ -41,7 +41,7 @@
             {% endfor %}
           {% else %}
-            <p> This group has no pending invitations.</p>
+            <p class="text-muted">This group has no pending invitations.</p>
```
``text-muted`` looks OK. I suggest you also add a CSS rule like the following (with a more appropriate selector):
```
form.edit-group p {
    text-align: center;
}
```

comzeradd replied:
I'm not sure about the alignment. We have no text aligned in center. I think it will look a bit odd.

johngian replied:
It looks a bit odd to me now. It feels like its a common layout to have centered informative text when a list is empty.

comzeradd replied:
We don't use that kind of alignment nowhere else in the site, afaik. Only on alerts.

johngian replied:
Do we have a similar layout somewhere in Mozillians.org so we can follow the same paradigm?

johngian replied:
If not I suggest we go with the centered text.

comzeradd replied:
We have cases where data is pending (like secondary emails or api keys), but not a 100% similar case with this. We just have no centered text in any settings option or text.

johngian replied:
I suggest we go with the centered and let's reevaluate it in the future.

johngian commented:
New membership terms and membership expiration have different label text format (hint ``:``)

johngian commented:
I still suggest we consolidate terms/expiration. It doesn't really make any sense to have 2 different panels for these.

@akatsoulas what do you think?

comzeradd commented:
I agree. This would require mostly a backend forms consolidation, rather that frontend changes.

johngian commented:
I think that the way we handle forms right now in the backend it would be the exact same thing if we post these 2 fields seperately and together.

johngian commented on mozillians/jinja2/groups/includes/group_edit_access.html:
```diff
@@ -25,7 +25,7 @@
         {% endif %}
 
-        <fieldset id="access">
+        <fieldset id="access" class="last-fieldset">
```
Why do we need this class? Also why all that DOM magic when you could select the last fieldset using CSS selectors?

comzeradd replied:
The `:last-child` selector doesn't work here because this is not considered last child, even when the next fieldset is hidden.

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -18,8 +19,10 @@ $(function() {
     function checkCriteria(modVal) {
         if(modVal === 'by_request') {
             $id_new_member_criteria_fieldset.show();
+            $access.removeClass('last-fieldset');
```
Same here.

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -18,8 +19,10 @@ $(function() {
+            $access.removeClass('last-fieldset');
         } else {
             $id_new_member_criteria_fieldset.hide();
+            $access.addClass('last-fieldset');
```
Same here.

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -42,10 +45,11 @@ $(function() {
     // Hide/show field when moderation field changes
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $('#form-access input[type="radio"]').on('change', function() {
```
Please use a var for that. Ideally we should use an ``id``.

johngian commented:
Membership expiration js is also broken.

johngian commented:
New membership criteria textfield should be under the label like in the rest of the same fields

![selection_010](https://cloud.githubusercontent.com/assets/835321/13920855/bc171694-ef7d-11e5-8918-39af7960e419.png)

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -42,10 +45,11 @@ $(function() {
+        checkCriteria($accepting_new_members.val());
     });
-    $id_membership_can_expire.change(function() {
+    $accepting_new_members.change(function() {
```
This line here is wrong. You are checking group type to figure out if group membership expires.

akatsoulas replied:
Why is this changed in the first place?

comzeradd replied:
Because the previous code was broken. Terms input text is never revealed.

`$accepting_new_members` is `'input[name=accepting_new_members]:checked'`, which just checks which of all three radio buttons is checked.

johngian replied:
Again this is for the group type not for the membership expiration.

akatsoulas commented:
:+1: for merging the two sections into one.

comzeradd commented:
PR updated. Please test that everything works as expected, especially at the Access tab.

johngian commented:
Please reduce the space between the 2 checkboxes:
![selection_013](https://cloud.githubusercontent.com/assets/835321/13926842/c954bc08-ef96-11e5-989d-40bb466acf99.png)

johngian commented:
Also do we need these 2 labels to be bold? (no strong opinion on that)

johngian commented:
I preferred the previous UI with the choices after the label.
![selection_014](https://cloud.githubusercontent.com/assets/835321/13927637/2340a37c-ef9b-11e5-973c-e0aef505957c.png)

johngian commented:
Also you need to rebase and adapt the code after #1391 fixes.

johngian commented:
"Membership can expire" is not working.

* Clicked checkbox
* Added days

No data gets saved.

comzeradd commented:
I fixed the modal misplacement and rebased with master.

I reverted the change with joining terms and expiration. This doesn't make sense to be part of this PR. The forms are separated in the backend and even the tests have to be rewritten.

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -7,9 +7,10 @@ $(function() {
+    $accepting_new_members = $('input[name=accepting_new_members]:checked');
     $id_new_member_criteria_fieldset = $('#id_new_member_criteria_fieldset');
-    $id_invalidation_days = $('#id_group_invalidation_days');
+    $id_group_invalidation_days = $('#id_group_invalidation_days');
```
No need to rename the vars here.

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -42,8 +43,9 @@ $(function() {
     // Hide/show field when moderation field changes
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $($group_type_choice).on('change', function() {
```
No need to reselect the selection.

comzeradd commented:
@johngian PR updated

johngian commented on mozillians/static/mozillians/js/group_edit.js:
```diff
@@ -42,8 +43,9 @@ $(function() {
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $group_type_choice.on('change', function() {
+        $accepting_new_members = $('input[name=accepting_new_members]:checked');
```
Instead of reselecting this here you could use ``this``.

johngian replied:
Instead of copy/pasting the selector here you could reevaluate like:
``$accepting_new_members = $($accepting_new_members.selector)``

johngian commented:
Code looks OK. r+wc

johngian commented:
Please test it locally before you merge.
